lib/ansible/modules/crypto/openssl_dhparam.py

```diff
@@ -236,9 +344,31 @@ def main():
             msg="The directory '%s' does not exist or the file is not a directory" % base_dir
         )
 
-    dhparam = DHParameter(module)
-
-    if dhparam.state == 'present':
+    if module.params['state'] == 'present':
+        backend = module.params['select_crypto_backend']
+        if backend == 'auto':
+            # Detection what is possible
+            can_use_cryptography = CRYPTOGRAPHY_FOUND and CRYPTOGRAPHY_VERSION >= LooseVersion(MINIMAL_CRYPTOGRAPHY_VERSION)
+            can_use_openssl = module.get_bin_path('openssl', False) is not None
+
+            # First try cryptography, then OpenSSL
+            if can_use_cryptography:
+                backend = 'cryptography'
+            elif can_use_openssl:
+                backend = 'openssl'
+
+            # Success?
+            if backend == 'auto':
+                module.fail_json(msg=("Can't detect either the required Python library cryptography (>= {0}) "
+                                      "or the OpenSSL binary openssl").format(MINIMAL_CRYPTOGRAPHY_VERSION))
+
+        if backend == 'openssl':
```

Comment:
```suggestion
        elif backend == 'openssl':
```
I didn't check but looks logical here

Reply:
That would be wrong: in the `if` before it, the value of `backend` is set to `openssl` or `cryptography` (or `module.fail_json()` is called). So if this would be `elif`, the `dhparam` would end up uninitialized.

Reply:
clear, thanks